.github/workflows/python-package.yml

```diff
@@ -0,0 +1,54 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python
+
+name: Python package
+
+on: pull_request
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.12"]
+        poetry-version: ["1.8.3"]
+
+    permissions:
+      pull-requests: write
+    
+    steps:
+    - uses: actions/checkout@v4
+    
+    - name: "Setup Python, Poetry and Dependencies"
+      uses: packetcoders/action-setup-cache-python-poetry@main
```

Comment:
I'm a bit wary of this being `@main`, as it means any (potentially large) project updates could be pulled in without warning. Is there a disadvantage to [cacheing using the standard `setup-python`](https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#caching-packages) action? (Assuming we're only planning to test against the one poetry version, which I think would be fine.)

Reply:
I had come across issues [like this](https://github.com/actions/setup-python/issues/846), which made me unsure about using that approach (the issue mentioned there is the one noted under the poetry section in your link as well). But I do think that issue is unlikely to be relevant for us, so we can switch, or I could alternatively pin this action at the current version 1.2.